typescript/infra/test/govern.hardhat-test.ts

```diff
@@ -0,0 +1,217 @@
+      });
+    }
+  }
+}
+
+describe('ICA governance', async () => {
+  const localChain = Chains.test1;
+  const remoteChain = Chains.test2;
+
+  let signer: SignerWithAddress;
+  let multiProvider: MultiProvider;
+  let accountConfig: AccountConfig;
+  let coreApp: TestCoreApp;
+  // let local: InterchainAccountRouter;
+  let remote: InterchainAccountRouter;
+  let routerConfig: ChainMap<RouterConfig>;
+  let contracts: HyperlaneContractsMap<InterchainAccountFactories>;
+  let icaApp: InterchainAccount;
+  let recipient: TestRecipient;
+  let accountOwner: Address;
+  let governor: HyperlaneTestGovernor;
+
+  before(async () => {
+    [signer] = await ethers.getSigners();
+    multiProvider = MultiProvider.createTestMultiProvider({ signer });
+    const ismFactoryDeployer = new HyperlaneProxyFactoryDeployer(multiProvider);
+    const ismFactory = new HyperlaneIsmFactory(
+      await ismFactoryDeployer.deploy(multiProvider.mapKnownChains(() => ({}))),
+      multiProvider,
+    );
+
+    coreApp = await new TestCoreDeployer(multiProvider, ismFactory).deployApp();
+    routerConfig = coreApp.getRouterConfig(signer.address);
+  });
+
+  beforeEach(async () => {
+    contracts = await new InterchainAccountDeployer(multiProvider).deploy(
+      routerConfig,
+    );
+    // local = contracts[localChain].interchainAccountRouter;
+    remote = contracts[remoteChain].interchainAccountRouter;
+    icaApp = new InterchainAccount(contracts, multiProvider);
+
+    accountConfig = {
+      origin: Chains.test1,
+      owner: signer.address,
+      localRouter: remote.address,
+    };
+
+    const recipientF = new TestRecipient__factory(signer);
+    recipient = await recipientF.deploy();
+
+    const contractsMap = {
+      [remoteChain]: {
+        recipient,
+      },
+      [localChain]: {
+        recipient,
+      },
+    };
+    // missing ica
+    const configMap = {
+      [localChain]: { owner: signer.address },
+      [remoteChain]: {
+        owner: { origin: Chains.test1, owner: signer.address },
+      },
+    };
+
+    const app = new TestApp(contractsMap, multiProvider);
+    const checker = new TestChecker(multiProvider, app, configMap);
+    governor = new HyperlaneTestGovernor(checker, icaApp);
+
+    accountOwner = await resolveAccountOwner(
+      multiProvider,
+      remoteChain,
+      accountConfig,
+    );
+    await recipient.transferOwnership(accountOwner);
+  });
+
+  it('changes ISM on the remote recipient', async () => {
+    // precheck
+    const actualOwner = await recipient.owner();
+    expect(actualOwner).to.equal(accountOwner);
+
+    // arrange
+    const newIsm = randomAddress();
+    await governor.checker.checkChain(Chains.test2);
+    const call = {
+      to: recipient.address,
+      data: recipient.interface.encodeFunctionData(
+        'setInterchainSecurityModule',
+        [newIsm],
+      ),
+      value: BigNumber.from(0),
+      description: 'Setting ISM on the test recipient',
+    };
```

Comment:
pretty sure `recipient.populateTransaction.setInterchainSecurityModule` is your friend here